Core/Core/Features/Grades/View/GradeRowView.swift

```diff
@@ -64,9 +66,16 @@ public struct GradeRowView: View {
             let submission = assignment.submissions?.first { $0.userID == userID }
             let displayProperties = submission?.stateDisplayProperties ?? .usingStatus(.notSubmitted)
 
-            Text(displayProperties.text)
-                .foregroundStyle(Color(displayProperties.color))
-                .font(.regular14)
+            HStack(spacing: 2) {
+                Image(uiImage: displayProperties.icon)
+                    .resizable()
+                    .frame(width: uiScale.iconScale * 18, height: uiScale.iconScale * 18)
```

Comment:
We have a helper which could be used here:
```suggestion
                    .size(uiScale.iconScale * 18)
```
Same in other places where you use this